src/utils/storage/index.js

```diff
@@ -0,0 +1,17 @@
+
+// @ts-check
+
+import { indexedDbStorage } from './indexed-db-storage.js'
+import { memoryStorage } from './memory-storage.js'
+
+/**
+ * @typedef {object} Storage
+ * @property {function(string):Promise<any>} get - Retrieves the value associated with the key.
+ * @property {function(string,any):Promise<void>} set - Sets a new value for the key.
```

Comment:
its not serialised automatically. I guess it could be any value but we have to add some extra handling for some storage implementations. For example localStorage only accepts strings as keys and values so what ever gets passed has to be serialisable as a string.

Reply:
gonna merge then, if we need to add support for non-string storage keys we can do that.